pkg/controllers/dynakube/extension/databases/deployment.go

```diff
@@ -0,0 +1,254 @@
+package databases
+
+import (
+	"context"
+	"fmt"
+	"maps"
+	"slices"
+
+	"github.com/Dynatrace/dynatrace-operator/pkg/api/latest/dynakube"
+	"github.com/Dynatrace/dynatrace-operator/pkg/api/latest/dynakube/extensions"
+	"github.com/Dynatrace/dynatrace-operator/pkg/consts"
+	"github.com/Dynatrace/dynatrace-operator/pkg/util/kubeobjects/labels"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	probePort = int32(8080)
+	// Keep in sync with helm chart
+	defaultServiceAccount = "dynatrace-database-extensions-executor"
+	// Must contain the ID specified in the DynaKube CR.
+	executorLabelKey = "extensions.dynatrace.com/executor.id"
+	// Type of data source to allow EEC to separate them. For database extensions this is always "sql".
+	datasourceLabelKey   = "extensions.dynatrace.com/datasource"
+	datasourceLabelValue = "sql"
+
+	userDataVolumeName = "user-data"
+	userDataMountPath  = "/var/userdata"
+	tokenVolumeName    = "auth-token"
+	tokenMountPath     = "/var/run/dynatrace/executor/token"
+	certsVolumeName    = "https-certs"
+	certsMountPath     = "/certs"
+)
+
+func buildAllLabels(dk *dynakube.DynaKube, dbex extensions.DatabaseSpec) (map[string]string, map[string]string, map[string]string) {
+	appLabels := labels.NewAppLabels(labels.DatabaseDatasourceLabel, dk.Name, labels.DatabaseDatasourceLabel, "")
+
+	deploymentLabels := appLabels.BuildLabels()
+	// Remove empty version label
+	delete(deploymentLabels, labels.AppVersionLabel)
+
+	matchLabels := appLabels.BuildMatchLabels()
+
+	templateLabels := maps.Clone(deploymentLabels)
+	templateLabels[executorLabelKey] = dbex.ID
+	templateLabels[datasourceLabelKey] = datasourceLabelValue
+
+	if dbex.Labels != nil {
+		temp := maps.Clone(dbex.Labels)
+		maps.Copy(temp, templateLabels)
+		templateLabels = temp
+	}
+
+	return deploymentLabels, matchLabels, templateLabels
+}
+
+func buildServiceAccountName(dbex extensions.DatabaseSpec) string {
+	if dbex.ServiceAccountName != "" {
+		return dbex.ServiceAccountName
+	}
+
+	return defaultServiceAccount
+}
+
+func buildContainer(dk *dynakube.DynaKube, dbex extensions.DatabaseSpec) corev1.Container {
+	pullPolicy := corev1.PullIfNotPresent
+	if dk.Spec.Templates.DatabaseExecutor.ImageRef.Tag == "latest" {
+		// For initial testing latest image is used, so let runtime pull updates if they're available.
+		// Maybe move this into the imageRef, e.g. imageRef.PullPolicy()
```

Comment:
Please create a followup so we don't forget about this.

Reply:
done: DAQ-15372